_overviews/scala3-contribution/arch-symbols.md

```diff
@@ -0,0 +1,133 @@
+---
+title: Symbols
+type: section
+description: This page describes symbols in the Scala 3 compiler.
+num: 17
+previous-page: arch-time
+next-page: arch-context
+---
+
+> (The following is work in progress), adapted from dotty.epfl.ch
+
+## Symbols and SymDenotations
+
+ - why symbols are not enough: their contents change all the time
+ - reference: string + sig
+
+
+`dotc` is different from most other compilers in that it is centered around the idea of
+maintaining views of various artifacts associated with code. These views are indexed
+by time.
+
+A symbol refers to a definition in a source program. Traditionally,
+compilers store context-dependent data in a _symbol table_. The
+symbol then is the central reference to address context-dependent
+data. But for the requirements of `dotc` it turns out that symbols are
+both too little and too much for this task.
+
+**Too little:** The attributes of a symbol depend on the phase. Examples:
+Types are gradually simplified by several phases. Owners are changed
+in phases `LambdaLift` (when methods are lifted out to an enclosing
+class) and Flatten (when all classes are moved to top level). Names
+are changed when private members need to be accessed from outside
+their class (for instance from a nested class or a class implementing
+a trait). So a functional compiler, a `Symbol` by itself met mean
```

Comment:
?

Reply:
this entire page was just copy pasted from the old dotty website, so I added the banner to say it is a work in progress at the top of the page

Reply:
same with the pages about contexts and periods